### PR TITLE
Add deployment guide for GLBC in non-gcp mode.

### PR DESCRIPTION
**Pull request by cxhiano**

### CODE REVIEW

File: docs/deploy/gke/non-gcp/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Overview
+
+This document will deploy the self managed Ingress-GCE controller in Non-GCP
```
Comment (bowei) [inline docs/deploy/gke/non-gcp/README.md]: you mean non-GKE

Comment (bowei) [reply docs/deploy/gke/non-gcp/README.md]: or something else...

File: docs/deploy/gke/non-gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ mode.
```
Comment (freehan) [inline docs/deploy/gke/non-gcp/README.md]: This document will deploy the self managed Ingress-GCE controller running on GCP in Non-GCP
 mode.

File: docs/deploy/gke/non-gcp/README.md (hunk 1 above)
Comment (cxhiano) [reply docs/deploy/gke/non-gcp/README.md]: Eventually we'd like deploy it on a non-GCP environment like GKE On-Prem. Currently we deploy it on GCP GKE to facilitate testing.

**Pull request merged**